### ISSUE TITLE
Crash on trying to use vanilla crate as ingredient in modded recipe

### ISSUE THREAD

**Issue by Craluminum2413** [anegostudios/VintageStory-Issues#1840]
**Game Version:** 1.16.5-rc.1 
**Platform:** Windows
**Modded:** Yes
**SP/MP:** Singleplayer

### Description
Example of a recipe that causing the crash:
```json
{
  "ingredientPattern": "T__,__C,_CC",
  "ingredients": {
    "T": { "type": "item", "code": "game:saw-*", "isTool": true },
    "C": { "type": "block", "code": "game:crate", "quantity": 9 }
  },
  "copyAttributesFrom": "C",
  "recipeGroup": 2,
  "width": 3,
  "height": 3,
  "output": { "type": "block", "code": "crate3x3x3-east", "attributes": { "lidState": "closed" } }
}
```

### How to reproduce
1. Use this mod [MultiblockCrates-v1.0.1.zip](https://github.com/anegostudios/VintageStory-Issues/files/8429130/MultiblockCrates-v1.0.1.zip)
2. Try to craft any big crate from the mod using vanilla crates and saw
3. It crashes when pattern matches

### Logs
```
Game Version: v1.16.5-rc.1 (Unstable)
Loaded Mods: game@1.16.5-rc.1, mcrate@1.0.1, creative@1.16.5-rc.1, survival@1.16.5-rc.1
04/06/2022 20:26:43: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.GridRecipe.<>c__DisplayClass23_0.<GetInputStackForPatternCode>b__0(GridRecipeIngredient ig)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Vintagestory.API.Common.GridRecipe.GetInputStackForPatternCode(String patternCode, ItemSlot[] inputSlots)
   at Vintagestory.Common.InventoryCraftingGrid.FindMatchingRecipe()
   at Vintagestory.API.Common.InventoryBase.DidModifyItemSlot(ItemSlot slot, ItemStack extractedStack)
   at Vintagestory.API.Common.ItemSlot.OnItemSlotModified(ItemStack sinkStack)
   at Vintagestory.API.Common.ItemSlot.ActivateSlotRightClick(ItemSlot sourceSlot, ItemStackMoveOperation& op)
   at Vintagestory.API.Common.InventoryBase.ActivateSlot(Int32 slotId, ItemSlot sourceSlot, ItemStackMoveOperation& op)
   at Vintagestory.Common.InventoryCraftingGrid.ActivateSlot(Int32 slotId, ItemSlot sourceSlot, ItemStackMoveOperation& op)
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.SlotClick(ICoreClientAPI api, Int32 slotId, EnumMouseButton mouseButton, Boolean shiftPressed, Boolean ctrlPressed, Boolean altPressed)
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.OnMouseMove(ICoreClientAPI api, MouseEvent args)
   at Vintagestory.API.Client.GuiComposer.OnMouseMove(MouseEvent mouse)
   at Vintagestory.API.Client.GuiDialog.OnMouseMove(MouseEvent args)
   at Vintagestory.Client.NoObf.GuiManager.OnMouseMove(MouseEvent args)
   at Vintagestory.Client.NoObf.ClientMain.OnMouseMove(MouseEvent args)
   at Vintagestory.Client.NoObf.ClientPlatformWindows.UpdateMousePosition()
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _TMlc0SVi1veOk3WmBa9fJGYHBkJ._nKpQGPZPKMHALS5qaTLoxqhCR9O(_KkN6emy6owjnJ1fsZJmPr1AZx0f , String[] )
   at _iv6cwviezE8MtOIBckvWYzXitAC._nKpQGPZPKMHALS5qaTLoxqhCR9O(ThreadStart )
```

**Comment by Craluminum2413** [comment]
Just found out it doesn't crash for crates from my mod by using exactly the same recipe, except instead game:crate I use crate2x1-east, maybe something is wrong with **domain** for copyAttributesFrom

**Comment by Craluminum2413** [comment]
Although no, I just found out that copyAttributesFrom supports only one attribute'd block per grid recipe

**Comment by Craluminum2413** [comment]
I found a workaround for that:
```json
{
  "ingredientPattern": "TE_,CC_,CC_",
  "ingredients": {
    "T": { "type": "item", "code": "game:saw-*", "isTool": true },
    "C": { "type": "block", "code": "crate3x1-*", "quantity": 2 },
    "E": { "type": "block", "code": "crate3x1-*" }
  },
  "copyAttributesFrom": "E",
  "recipeGroup": 2,
  "width": 3,
  "height": 3,
  "output": { "type": "block", "code": "crate3x3x3-east", "attributes": { "lidState": "closed" } }
}
```
Here is only one crate gives attributes, thus, no crashes
![image](https://user-images.githubusercontent.com/69315569/162177541-6706d46f-545d-457a-9a74-d7c36b139437.png)

**Comment by Craluminum2413** [comment]
Another very strange find:
This recipe works:
```json
{
  "ingredientPattern": "C_T,___,___",
  "ingredients": {
    "T": { "type": "item", "code": "game:saw-*", "isTool": true },
    "C": { "type": "block", "code": "game:crate", "quantity": 3 }
  },
  "copyAttributesFrom": "C",
  "recipeGroup": 2,
  "width": 3,
  "height": 3,
  "output": { "type": "block", "code": "crate3x1-east", "attributes": { "lidState": "closed" } }
}
```
While this doesn't:
```json
{
  "ingredientPattern": "T__,___,_C_",
  "ingredients": {
    "T": { "type": "item", "code": "game:saw-*", "isTool": true },
    "C": { "type": "block", "code": "game:crate", "quantity": 18 }
  },
  "copyAttributesFrom": "C",
  "recipeGroup": 2,
  "width": 3,
  "height": 3,
  "output": { "type": "block", "code": "crate3x3x2-east", "attributes": { "lidState": "closed" } }
}

```

**Comment by tyronx** [comment]
strangely enough, it seems to just ignore my crafting attempt
<img width="185" alt="2022-04-10_19-51-20" src="https://user-images.githubusercontent.com/762006/162630516-5051f91a-1325-4ecc-8a18-37c481a26576.png">
s

**Comment by Craluminum2413** [comment]
Can you craft 3x1 and 2x1 crates?

**Comment by tyronx** [comment]
Yes, I looked at th recipe files. I think the 3x3x recipe is missing. I added it, now it crashes when trying to craft. Strange that this never happened before. 

Will be fixed in 1.16.5 👍
